server/src/test/java/org/opensearch/search/SearchIpFieldTermsTests.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search;
+
+import org.opensearch.action.bulk.BulkRequestBuilder;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.common.network.InetAddresses;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.test.OpenSearchSingleNodeTestCase;
+import org.hamcrest.MatcherAssert;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.opensearch.action.support.WriteRequest.RefreshPolicy.IMMEDIATE;
+import static org.hamcrest.Matchers.equalTo;
+
+public class SearchIpFieldTermsTests extends OpenSearchSingleNodeTestCase {
+
+    public static final boolean IPv4_ONLY = true;
+    static String defaultIndexName = "test";
+
+    public void testMassive() throws Exception {
+        XContentBuilder xcb = createMapping();
+        client().admin().indices().prepareCreate(defaultIndexName).setMapping(xcb).get();
+        ensureGreen();
+
+        BulkRequestBuilder bulkRequestBuilder = client().prepareBulk();
+
+        int cidrs = 0;
+        int ips = 0;
+        List<String> toQuery = new ArrayList<>();
+        for (int i = 0; ips <= 1024 && i < 1000000; i++) {
+            final String ip;
+            final int prefix;
+            if (IPv4_ONLY) {
+                ip = generateRandomIPv4();
```

Comment:
it's just IP4 test, no IP6. Is IP6 necessary, is it worth to combine 6th and 4th in the single test?